Broken PR. Will be deleted (#9924)
ydb-platform/ydb: ydb/docs/en/core/dev/troubleshooting/schemas/overloaded-shards.md

# Overloaded shards

## Description

> [Data shards](../../../concepts/glossary.md#data-shard) in {{ ydb-short-name }} might get overloaded for the following reasons:

Reviewer: This article discusses several related issues; it probably makes sense to adjust the common article structure for this case.

Currently:

```text
# Header

## Description
...

- Issue 1
- Issue 2

## Diagnostics

...

## Recommendations

- Issue 1
- Issue 2
```

Could be:

```text
# Header

Introduction...

- [Issue 1](#issue-1)
- [Issue 2](#issue-2)

## Issue 1 {#issue-1}

More details on the issue.

### Diagnostics

How to check whether it applies or not.

### Recommendations

...

## Issue 2 {#issue-2}

More details on the issue.

### Diagnostics

How to check whether it applies or not.

### Recommendations

...

```

Author: We decided on a slightly different structure here with different recommendations for each case

* A table is created without the [AUTO_PARTITIONING_BY_LOAD](../../../concepts/datamodel/table.md#AUTO_PARTITIONING_BY_LOAD) clause.

>     In this case {{ ydb-short-name }} does not split overloaded shards. If a shard has to process too many queries, it will use up all its CPU resources and, consequently, query response times will increase.

Reviewer: It's not necessarily about CPU resources; it's about the fact that data shards are single-threaded and might not keep up with whatever is requested from them. Also, I'd describe this more carefully as there are a bit different cases: as far as I understand, `OVERLOADED` errors are returned more or less instantly when the load thresholds are exceeded (but you'll still lose time on retries), while the latency would be higher for queries that got accepted and wait for their turn to be executed (but they won't get the `OVERLOADED` error).

> * A table is created with the [AUTO_PARTITIONING_MAX_PARTITIONS_COUNT](../../../concepts/datamodel/table.md#AUTO_PARTITIONING_MAX_PARTITIONS_COUNT) clause and has already reached the limit.

Reviewer: ```suggestion
* A table was created with the [AUTO_PARTITIONING_MAX_PARTITIONS_COUNT](../../../concepts/datamodel/table.md#AUTO_PARTITIONING_MAX_PARTITIONS_COUNT) setting and has already reached the partition limit.
```

Reviewer: In SQL standard, queries consist of *statements* separated by semicolons (like `SELECT` statement, `INSERT INTO` statement, etc.), and *clauses* are syntactic parts of statements (`GROUP BY` clause, `ORDER BY` clause, etc.). So, in this case, there would be a `WITH` clause, and these `AUTO_PARTITIONING_MAX_PARTITIONS_COUNT` are more like settings for it.

## Diagnostics

{% include notitle [#](_includes/overloaded-shards-diagnostics.md) %}

> ## Recommendations
> 
> Consider the following solutions to the shard overload issue:
> 
> * If the problem table is not partitioned by load, enable partitioning by load.
> 
> * If the table has reached the maximum number of partitions, increase the limit.

Reviewer: ```suggestion
## Recommendations

Consider the following solutions to address the shard overload issue:

* If the problem table is not partitioned by load, enable partitioning by load.

* If the table has reached the maximum number of partitions, increase the partition limit.
```
